### PR TITLE
Query for designation, department, employee search

### PR DESCRIPTION
**Pull request by nohan-antony**
3 queries added 
frontend not completed

### CODE REVIEW

File: src/services/WolfDen.API/Controllers/Department/Department.cs [hunk 1]
```diff
@@ -12,10 +14,16 @@ public class Department(IMediator mediator) : ControllerBase
         private readonly IMediator _mediator=mediator;
 
         
-        [HttpPost]
+        [HttpPost("add-department")]
```
Comment (Aravindh-sreenivas) [inline src/services/WolfDen.API/Controllers/Department/Department.cs:17]: Ideally this is not required, [POST] request in a given controller usually denotes the creation/ adding of the Entity related to the controller.
Is there any specific reason why `add-department` was added?

File: src/services/WolfDen.API/Controllers/Department/Department.cs [hunk 2]
```diff
@@ -12,10 +14,16 @@ public class Department(IMediator mediator) : ControllerBase
...
         public async Task<int> AddDepartment([FromBody] AddDepartmentCommand command,CancellationToken cancellationToken)
         {
             return await _mediator.Send(command,cancellationToken);
         }
+
+        [HttpGet("all")]
```
Comment (Aravindh-sreenivas) [inline src/services/WolfDen.API/Controllers/Department/Department.cs:23]: Similar to the previous comment, `all` is not required here.
As per norms, GET request denotes fetching the list of items, and GET("{id}") is more specific and implies fetching individual items.

File: src/services/WolfDen.API/Controllers/Designation/Designation.cs [hunk 3]
```diff
@@ -16,5 +19,11 @@ public async Task<int> AddDesignation([FromBody] AddDesignationCommand command,C
         {
             return await _mediator.Send(command,cancellationToken);
         }
+
+        [HttpGet("all-designation")]
```
Comment (Aravindh-sreenivas) [inline src/services/WolfDen.API/Controllers/Designation/Designation.cs:23]: See previous comments

File: src/services/WolfDen.Application/Requests/Queries/Employees/EmployeeDirectory/GetAllEmployeeQueryHandler.cs [hunk 4]
```diff
@@ -52,13 +56,13 @@ public async Task<PaginationResponse> Handle(GetAllEmployeeQuery request, Cancel
                    IsActive = e.IsActive
                })
             .ToListAsync(cancellationToken);
-            int totalpage = employees.Count;
-            int pageCount = (int)(Math.Ceiling((decimal)totalpage / request.PageSize));
-            var emp = employees.Skip(pageSize * pageNumber)
-               .Take(pageSize).ToList();
+            /* int totalpage = employees.Count;*/
```
Comment (Aravindh-sreenivas) [inline src/services/WolfDen.Application/Requests/Queries/Employees/EmployeeDirectory/GetAllEmployeeQueryHandler.cs:59]: Remove commented code if not needed.
Revisit all files to remove unwanted whitespaces and usings
- CTRL + R + CTRL + G : Removes unused usings
- CTRL + K+ CTRL + D : Formats document

File: src/services/WolfDen.Application/Requests/Queries/Employees/GetAllEmployeesName/GetAllEmployeesNameQueryHandler.cs [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    public class GetAllEmployeesNameQueryHandler : IRequestHandler<GetAllEmployeesNameQuery, List<EmployeeNameDTO>>
+    {
+        private readonly WolfDenContext _context;
+
+        public GetAllEmployeesNameQueryHandler(WolfDenContext context)
+        {
+            _context = context;
+        }
+
+        public async Task<List<EmployeeNameDTO>> Handle(GetAllEmployeesNameQuery query, CancellationToken cancellationToken)
```
Comment (Aravindh-sreenivas) [inline src/services/WolfDen.Application/Requests/Queries/Employees/GetAllEmployeesName/GetAllEmployeesNameQueryHandler.cs:17]: `GetAllEmployeesByNameQuery`

**Review by Aravindh-sreenivas — APPROVED** [review]

File: src/services/WolfDen.API/Controllers/Department/Department.cs (hunk 2 above)
Comment (nohan-antony) [reply src/services/WolfDen.API/Controllers/Department/Department.cs:23]: this api only returns name and id inorder to assign manager in front end
 [HttpGet("all")] will return all employee details for employee directory

File: src/services/WolfDen.API/Controllers/Designation/Designation.cs (hunk 3 above)
Comment (nohan-antony) [reply src/services/WolfDen.API/Controllers/Designation/Designation.cs:23]: done

File: src/services/WolfDen.Application/Requests/Queries/Employees/EmployeeDirectory/GetAllEmployeeQueryHandler.cs (hunk 4 above)
Comment (nohan-antony) [reply src/services/WolfDen.Application/Requests/Queries/Employees/EmployeeDirectory/GetAllEmployeeQueryHandler.cs:59]: done

File: src/services/WolfDen.Application/Requests/Queries/Employees/GetAllEmployeesName/GetAllEmployeesNameQueryHandler.cs (hunk 5 above)
Comment (nohan-antony) [reply src/services/WolfDen.Application/Requests/Queries/Employees/GetAllEmployeesName/GetAllEmployeesNameQueryHandler.cs:17]: done

File: src/ui/src/app/Service/leave-management.service.ts [hunk 6]
```diff
@@ -19,37 +19,32 @@ import { IEditLeaveType } from '../Interface/edit-leave-type'
...
-    return this.http.get<Array<IGetLeaveTypeIdAndname>>("https://localhost:7015/api/leave-type")
-  }
+    updateLeaveSettings(updateLeaveSettings : FormGroup<IUpdateLeaveSetting> ){
+      return this.http.put<boolean>("https://localhost:7015/api/leave-setting",updateLeaveSettings.value)
+    }
 
-  editLeaveType(editLeaveType: FormGroup<IEditLeaveType>) {
-    return this.http.put('https://localhost:7015/api/leave-type', editLeaveType.value);
-  }
-
```
[the remark this answers is not in the document]
Comment (nohan-antony) [reply src/ui/src/app/Service/leave-management.service.ts:51]: frontend corrected

File: src/ui/src/app/app.routes.ts [hunk 7]
```diff
@@ -70,9 +72,14 @@ export const routes: Routes = [
 
 
     {
-        path:'dashboard',
+        path:'portal',
         component:DashboardComponent,
         children:[
+            { path: '', redirectTo: '/dashboard', pathMatch: 'full' }, 
+            {path:'dashboard',component:MainPageComponent,canActivate: [guardsGuard]},
+            {path:'employee-directory',component: EmployeeDirectoryComponent,canActivate: [guardsGuard]},
```
[the remark this answers is not in the document]
Comment (nohan-antony) [reply src/ui/src/app/app.routes.ts:80]: middle of doing it

File: src/services/WolfDen.API/Controllers/Department/Department.cs (hunk 1 above)
Comment (nohan-antony) [reply src/services/WolfDen.API/Controllers/Department/Department.cs:17]: no, corrected

**Pull request merged**